codegen/type_converter.go

```diff
@@ -789,33 +933,6 @@ func (f *FieldAssignment) IsTransform() bool {
 	return f.FromIdentifier[in:] != f.ToIdentifier[out:]
 }
 
-// checkOptionalNil nil-checks fields that may not be initialized along
-// the assign path
-func checkOptionalNil(
```

Comment:
I think this problem still exists. 

This was added to make sure that optional structs dont end up with go default values thus corrupting a JSON wire representation from `{ foo: 'bar' }` to `{ foo: 'bar', nestedThing: { a: '', b: 0 }`

In the case that the thrift is something like

```
struct Outer {
  1: required string foo
  2: optional Inner nestedThing
}

struct InnerThing {
  1: optional string a
  2: optional i32 b
}
```